src/azul/indexer/log_forwarding_controller.py

```diff
@@ -0,0 +1,36 @@
+import json
+
+import chalice.app
+
+from azul import (
+    cached_property,
+)
+from azul.chalice import (
+    AppController,
+)
+from azul.indexer.log_forwarding_service import (
+    LogForwardingService,
+)
+
+
+class LogForwardingException(Exception):
+    pass
+
+
+class LogForwardingController(AppController):
+    """
+    Forward logs from an Application Load Balancer (ALB) to standard output.
+    When this behavior is invoked via a Lambda, the output is forwarded to the
+    default CloudWatch log group associated with the Lambda.
+    """
+
+    @cached_property
+    def service(self) -> LogForwardingService:
+        return LogForwardingService()
+
+    def forward_logs(self, event: chalice.app.S3Event) -> None:
+        try:
+            for message in self.service.read_logs(event.bucket, event.key):
+                print(json.dumps(message))
+        except Exception:
+            raise LogForwardingException(event.bucket, event.key)
```

Comment:
This might be useful reading. 

https://docs.aws.amazon.com/lambda/latest/dg/invocation-async.html

The reason I am asking is that I don't want Lambda to retry many times, if at all. I just want the degraded potentially hard to parse of the S3 log lines in the object to occur in the Lambda's CloudWatch logs.

Reply:
Ah, the lack of chalice support is not relevant since error-handling configuration is handled via a separate resource from the lambda function itself: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lambda_function_event_invoke_config#error-handling-configuration